lib/Dialect/LLHD/Transforms/TemporalRegions.h

```diff
@@ -20,33 +20,37 @@
 namespace circt {
 namespace llhd {
 
-struct TemporalRegionAnalysis {
-  using BlockMapT = DenseMap<Block *, int>;
-  using TRMapT = DenseMap<int, SmallVector<Block *, 8>>;
-
+class TemporalRegionAnalysis {
+public:
   explicit TemporalRegionAnalysis(Operation *op) { recalculate(op); }
 
+public:
   void recalculate(Operation *);
 
-  unsigned getNumTemporalRegions() { return numTRs; }
-
   int getBlockTR(Block *);
   SmallVector<Block *, 8> getBlocksInTR(int);
-
   SmallVector<Block *, 8> getExitingBlocksInTR(int);
+  SmallVector<int, 8> getTRSuccessors(int);
   Block *getTREntryBlock(int);
+
+public:
   bool hasSingleExitBlock(int tr) {
     return getExitingBlocksInTR(tr).size() == 1;
   }
+
   bool isOwnTRSuccessor(int tr) {
     auto succs = getTRSuccessors(tr);
     return std::find(succs.begin(), succs.end(), tr) != succs.end();
   }
 
-  SmallVector<int, 8> getTRSuccessors(int);
+  unsigned getNumTemporalRegions() { return numTRs; }
   unsigned getNumTRSuccessors(int tr) { return getTRSuccessors(tr).size(); }
   unsigned numBlocksInTR(int tr) { return getBlocksInTR(tr).size(); }
 
+public:
+  using BlockMapT = DenseMap<Block *, int>;
+  using TRMapT = DenseMap<int, SmallVector<Block *, 8>>;
+
```

Comment:
What's the motivation of reordering the type declarations and methods? I don't see why this order is better than the current one. Keep in mind that unnecessary reordering just adds unwanted churn.

Reply:
Since these type declarations are referred in declarations of private member variable only(maybe used somewhere outside the class, I guess), I got them together. If you don't like, I'll roll back this change. Thanks for your suggestion.